### ISSUE TITLE
Failure tolerance for matrix items

### ISSUE THREAD

**Issue by cevich** [cirruslabs/cirrus-ci-docs#78]
Given a matrix task across several platforms, I'd like a task to be fault-tolerant on one more images.  For example:

```yaml
testing_task:
    gce_instance:
        matrix:
            image_name: foo-1.2.3
            image_name: bar-2.3.4
    allow_failures: image =~ "foo.*"
    ...cut...
...cut...
```

Such that a task would be generated for each image, but in this example only the status of `bar-2.3.4` would apply.

**Comment by fkorotkov** [comment]
Hey @cevich, what if it will be possible to use an environment variable as an image name? Then you'll be able to do something like:

```
testing_task:
  env:
  matrix:
    IMAGE: foo-1.2.3
    IMAGE: bar-2.3.4
  gce_instance:
    image_name: $IMAGE
  allow_failures: $IMAGE =~ "foo.*"
```

Will it work for you?

**Comment by cevich** [comment]
YES, in fact that would be better!  It would allow our scripts to use that information rather than having to discover it redundantly.  Thanks!

**Comment by fkorotkov** [comment]
Fixed in ba84f2456e11d5a1c5a5b869b9c210cf5c2e31b2
